lib/src/database.dart

```diff
@@ -152,6 +152,39 @@ class Query {
         .then((snapshot) => new DataSnapshot(snapshot));
   }
 
+  /// Listens for data changes at a particular location.
+  ///
+  /// This is the primary way to read data from a [Database]. Your callback will
+  /// be triggered for the initial data and again whenever the data changes.
+  /// Use [off] to stop receiving updates.
+  ///
+  /// return function to unsubscribe
+  Function on<T>(
+      String eventType, Function(DataSnapshot<T> snapshot) callback) {
```

Comment:
According to the docs `callback` should have second optional `String` argument. Any reason you decided not to include it?

Also the `on` method itself should have 3rd optional argument for `cancelCallback`, right? (It should be safe to ignore the 4th argument for `context` as it seems JS specific).

Looking at the docs here: https://firebase.google.com/docs/reference/admin/node/admin.database.Query#on

Reply:
When using firebase in JS, I never really used the other two parameters, so wasn't paying much attention sry. Also I'm not very sure where to find the Error class. I included the third parameter only as a Function in the new commit.